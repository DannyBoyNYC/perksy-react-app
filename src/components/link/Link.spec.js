const React = require('react');
const PropTypes = require('prop-types');

const Link = require('./index').default;

describe('<Link.* />', () => {
  let component;
  const requiredProps = {
    children: <a>Text</a>,
    href: '#'
  };

  describe('<Link.Next />', () => {
    beforeEach(() => {
      const options = {
        context: {
          router: {
            prefetch: () => {},
            route: '/mock-route',
            pathname: 'mock-path'
          }
        },
        childContextTypes: {
          router: PropTypes.object
        }
      };
      component = shallowUntilTarget(
        <Link.Next {...requiredProps} />,
        'Link',
        options
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
