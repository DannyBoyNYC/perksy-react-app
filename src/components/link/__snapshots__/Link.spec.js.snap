// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link.* /> <Link.Next /> should render correctly with all required props 1`] = `
<Link
  href="#"
>
  <a
    className="withRouter(Link)-root-0-1-1"
  >
    Text
  </a>
</Link>
`;
